### PR TITLE
Add dualtor TSA/B/C support

### PR DESCRIPTION
**Pull request by lolyu**
#### Why I did it
Add TSA/B/C dualtor support

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

#### How I did it
For TSA, toggle all the mux to standby if the device type is `dualtor` and there are active mux ports.
For TSC, add mux status output.

#### How to verify it
Run TSA/B/C on a dualtor setup

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106

### CODE REVIEW

File: dockers/docker-fpm-frr/base_image_files/TSA [hunk 1]
```diff
@@ -1,3 +1,14 @@
 #!/bin/bash
 
+# toggle the mux to standby if dualtor and any mux active
+[[ "$(sonic-cfggen -d -v DEVICE_METADATA.localhost.subtype | tr [:upper:] [:lower:])" == *"dualtor"* ]]
```
Comment (yxieca) [inline dockers/docker-fpm-frr/base_image_files/TSA:4]: Suggest change this and line 6 to if statements/blocks.

File: dockers/docker-fpm-frr/base_image_files/TSA [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+is_dualtor=$?
+[[ $(show mux status | grep active | wc -l) > 0 ]]
+any_active=$?
+
+if [[ (${is_dualtor} -eq 0) && (${any_active} -eq 0) ]];
+then
```
Comment (yxieca) [inline dockers/docker-fpm-frr/base_image_files/TSA:10]: Add a log stating the toggle is happenning?

**Comment by zjswhhh** [comment]
Do we want to config the tor back to "auto" mode? When should we do that? Should we do it in TSB?

File: dockers/docker-fpm-frr/base_image_files/TSA (hunk 1 above)
Comment (lolyu) [reply dockers/docker-fpm-frr/base_image_files/TSA]: Re-formatted the conditionals

File: dockers/docker-fpm-frr/base_image_files/TSA (hunk 2 above)
Comment (lolyu) [reply dockers/docker-fpm-frr/base_image_files/TSA:7]: Added, thanks~

**Comment by lolyu** [comment]
> Do we want to config the tor back to "auto" mode? When should we do that? Should we do it in TSB?

Good point, added

**Review by zjswhhh — APPROVED** [review]
Changes look good to me.

File: dockers/docker-fpm-frr/base_image_files/TSA (hunk 2 above)
Comment (yxieca) [reply dockers/docker-fpm-frr/base_image_files/TSA:7]: Did you test the code? I think you introduced syntax error.

Comment (lolyu) [reply dockers/docker-fpm-frr/base_image_files/TSA:7]: Yes, tested with `set -x`
```
++ tr '[:upper:]' '[:lower:]'
++ sonic-cfggen -d -v DEVICE_METADATA.localhost.subtype
+ [[ dualtor == *\d\u\a\l\t\o\r* ]]
++ show mux status
++ wc -l
++ grep active
+ [[ 24 > 0 ]]
+ logger -t TSA -p user.info 'Toggle all mux mode to standby'
+ sudo config mux mode standby all
```

Comment (lolyu) [reply dockers/docker-fpm-frr/base_image_files/TSA:7]: Yes, tested with `set -x`
1. on a dualtor
```
++ sonic-cfggen -d -v DEVICE_METADATA.localhost.subtype
++ tr '[:upper:]' '[:lower:]'
+ [[ dualtor == *\d\u\a\l\t\o\r* ]]
++ grep active
++ wc -l
++ show mux status
+ [[ 24 > 0 ]]
+ logger -t TSA -p user.info 'Toggle all mux mode to standby'
+ sudo config mux mode standby all
```
2. on a t0
```
++ sonic-cfggen -d -v DEVICE_METADATA.localhost.subtype
++ tr '[:upper:]' '[:lower:]'
+ [[ '' == *\d\u\a\l\t\o\r* ]]
```

**Review by yxieca — APPROVED** [review]

**Pull request merged**